sentry-android-core/src/test/java/io/sentry/android/core/ConnectivityCheckerTest.kt

```diff
@@ -1,31 +1,161 @@
 package io.sentry.android.core
 
 import android.content.Context
-import androidx.test.core.app.ApplicationProvider
-import androidx.test.ext.junit.runners.AndroidJUnit4
+import android.content.pm.PackageManager.PERMISSION_DENIED
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkCapabilities.TRANSPORT_CELLULAR
+import android.net.NetworkCapabilities.TRANSPORT_ETHERNET
+import android.net.NetworkCapabilities.TRANSPORT_WIFI
+import android.net.NetworkInfo
+import android.os.Build
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
 import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
 import io.sentry.android.core.util.ConnectivityChecker
-import kotlin.test.BeforeTest
 import kotlin.test.Test
-import org.junit.runner.RunWith
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
 
-@RunWith(AndroidJUnit4::class)
 class ConnectivityCheckerTest {
 
-    private lateinit var context: Context
+    @Test
+    fun `When network is active and connected, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
```

Comment:
`m`: Why not extract the mock setup to `@Before`? Same could be done for `val net = mock<Network>()` and `val nc = mock<NetworkCapabilities>()`.

Reply:
because those mocks are not used in all the tests, using `Before` would slow down tests that are not being used at all